### PR TITLE
Fix small overhead in argument parser for strings

### PR DESCRIPTION
**Pull request by TheNormalnij**
`lua_strlen` looks unnecessary.

`lua_tolstring` returns string length in third argument.
`lua_tostring` is macro `lua_tolstring(L, (i), NULL)`

### CODE REVIEW

File: Shared/mods/deathmatch/logic/lua/LuaBasic.cpp [hunk 1]
```diff
@@ -15,17 +15,19 @@ namespace lua
...
 
     template <>
     std::string_view PopPrimitive<std::string_view>(lua_State* L, int& index)
     {
-        uint             uiLength = lua_strlen(L, index);
-        std::string_view outValue(lua_tostring(L, index++), uiLength);
+        size_t           uiLength;
+        const char*      str = lua_tolstring(L, index++, &uiLength);
+        std::string_view outValue(str, uiLength);
         return outValue;
```
Comment (Pirulax) [inline Shared/mods/deathmatch/logic/lua/LuaBasic.cpp:30]: Could just do
`return {str, uiLength}`

File: Shared/mods/deathmatch/logic/lua/LuaBasic.cpp [hunk 2]
```diff
@@ -15,17 +15,19 @@ namespace lua
     template <>
     std::string PopPrimitive<std::string>(lua_State* L, int& index)
     {
-        uint        uiLength = lua_strlen(L, index);
+        size_t      uiLength;
+        const char* str = lua_tolstring(L, index++, &uiLength);
         std::string outValue;
-        outValue.assign(lua_tostring(L, index++), uiLength);
+        outValue.assign(str, uiLength);
         return outValue;
```
Comment (Pirulax) [inline Shared/mods/deathmatch/logic/lua/LuaBasic.cpp]: Could just do
`return {PopPrimitive<std::string_view>(L, index)}`
(You will have to move the function definition below the string_view one for this to work)

File: Shared/mods/deathmatch/logic/lua/LuaBasic.cpp (hunk 1 above)
Comment (TheNormalnij) [reply Shared/mods/deathmatch/logic/lua/LuaBasic.cpp:30]: Prefer readable variant over compact.
`return std::string_view(str, uiLength);` has more sense

File: Shared/mods/deathmatch/logic/lua/LuaBasic.cpp (hunk 2 above)
Comment (TheNormalnij) [reply Shared/mods/deathmatch/logic/lua/LuaBasic.cpp]: same

File: Shared/mods/deathmatch/logic/lua/LuaBasic.cpp (hunk 1 above)
Comment (Pirulax) [reply Shared/mods/deathmatch/logic/lua/LuaBasic.cpp:30]: Fair enough, do that then

File: Shared/mods/deathmatch/logic/lua/LuaBasic.cpp (hunk 2 above)
Comment (Pirulax) [reply Shared/mods/deathmatch/logic/lua/LuaBasic.cpp]: So perhaps `return std::string{PopPrimitive<std::string_view>(L, index)}`?

**Comment by TheNormalnij** [comment]
fixed

**Review by ghost — APPROVED** [review]
LGTM :shipit:

File: Shared/mods/deathmatch/logic/lua/LuaBasic.cpp (hunk 2 above)
Comment (tederis) [inline Shared/mods/deathmatch/logic/lua/LuaBasic.cpp:21]: `return std::string(str, uiLength);` would be neater

Comment (TheNormalnij) [reply Shared/mods/deathmatch/logic/lua/LuaBasic.cpp:21]: Thx

**Comment by Dutchman101** [comment]
Reviewed by 3 people, and all code reviews were addressed; ready to merge

**Pull request merged**